src/main/java/formflow/library/data/SubmissionRepositoryService.java

```diff
@@ -116,32 +116,38 @@ public void removeSubflowCSRF(Submission submission, String subflowName) {
      */
     public synchronized void generateAndSetUniqueShortCode(Submission submission) {
 
+        if (shortCodeConfig.getConfig(submission.getFlow()) == null) {
+            return;
+        }
+
         if (submission.getShortCode() != null) {
             log.warn("Unable to create short code for submission {}", submission.getId());
             return;
         }
 
         log.info("Attempting to create short code for submission {}", submission.getId());
 
+        ShortCodeConfig.Config config = shortCodeConfig.getConfig(submission.getFlow());
+
         // If there is no short code for this submission in the database, generate one
-        int codeLength = shortCodeConfig.getCodeLength();
+        int codeLength = config.getCodeLength();
```

Comment:
Is it possible to get to this point with a null config?

Reply:
It shouldn't be! I just reorganized things slightly so the initial null check happens right before this point.